app/models/user.rb

```diff
@@ -17,6 +17,20 @@ def authorized_to?(access_request)
     end
   end
 
+  def inventory_groups
+    inventory_groups_permission = rbac_permissions.select do |permission|
+      permission.permission == Rbac::INVENTORY_GROUPS_READ
+    end
+
+    return Rbac::INVENTORY_GROUPS_GLOBAL if inventory_groups_permission.any? do |permission|
+      permission.resource_definitions == Rbac::INVENTORY_GROUPS_GLOBAL
+    end
+
+    inventory_groups_permission.collect do |permission|
```

Comment:
There are some edge-cases that this code would not handle well.

Access to all hosts even though there is a resourceDefinition:
```json
   {
     "resourceDefinitions": [
       {
         "attributeFilter": {
           "key": "group.id",
           "value": [],
           "operation": "in"
         }
       }
     ],
     "permission": "inventory:hosts:read"
   },
```

The key in the attribute filter is not verified:
```json
   {
     "resourceDefinitions": [
       {
         "attributeFilter": {
           "key": "foo.id",
           "value": [1,2,3,4,...],
           "operation": "in"
         }
       }
     ],
     "permission": "inventory:hosts:read"
   },
```

The operation can be `equal` as @coderbydesign said:
```json
   {
     "resourceDefinitions": [
       {
         "attributeFilter": {
           "key": "group.id",
           "value": 1234,
           "operation": "equal"
         }
       }
     ],
     "permission": "inventory:hosts:read"
   },
```